### PR TITLE
Add test cases for generate_reports()

### PR DESCRIPTION
**Pull request by JCantu248**
#  #

## 🗣 Description ##
This PR adds test cases for generate_reports() in tpt_reports.py. Resolves Issue #22 

## 💭 Motivation and context ##
The added test cases will test ensure that negative boolean status values will be returned if load_json_file() returns a null JSON object, or if parse_json() was not able to parse out valid fields from the file and returns null objects. 

## 🧪 Testing ##

Testing was performed by running PyTest and ensured all added and pre-existing test cases were passing.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by schmelz21 — APPROVED** [review]

File: tests/test_tpt_reports.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 RELEASE_TAG = os.getenv("RELEASE_TAG")
 PROJECT_VERSION = tpt_reports.__version__
 TEST_JSON_FILE = "tests/data/test.json"
+TEST_BAD_JSON_FILE = "tests/data/bad_file.json"
 DEFAULT_OUTPUT_DIRECTORY = "~/"
```
Comment (dav3r) [inline tests/test_tpt_reports.py]: I would alphabetize these now that we have a growing set of them:
```suggestion
DEFAULT_OUTPUT_DIRECTORY = "~/"
PROJECT_VERSION = tpt_reports.__version__
RELEASE_TAG = os.getenv("RELEASE_TAG")
TEST_BAD_JSON_FILE = "tests/data/bad_file.json"
TEST_JSON_FILE = "tests/data/test.json"
```

File: tests/test_tpt_reports.py [hunk 2]
```diff
@@ -143,6 +144,40 @@ def test_domain_validation(mock_generate_reports):
...
+@patch("tpt_reports.tpt_reports.report_gen")
+def test_generate_reports_bad_file_name(mock_report_gen):
+    """Validate functionality of generate_reports() when a bad filename is provided."""
+    mock_report_gen.return_value = False
+    return_val = tpt_reports.tpt_reports.generate_reports(
+        "test", "test", "cisa.gov", "./test_output", "nonexistent_file.json"
+    )
+    assert (
+        return_val is False
+    ), "generate_reports() returned a boolean True when a bad filename was provided."
```
Comment (dav3r) [inline tests/test_tpt_reports.py]: I'd reword the assertion error message slightly, since you are only validating that the return value is not `False` - there's no guarantee that this means it returned `True`:
```suggestion
    ), "generate_reports() failed to return a boolean False when a non-existent filename was provided."
```

File: tests/test_tpt_reports.py [hunk 3]
```diff
@@ -143,6 +144,40 @@ def test_domain_validation(mock_generate_reports):
...
+
+
+@patch("tpt_reports.tpt_reports.report_gen")
+def test_generate_reports_bad_file_data(mock_report_gen):
+    """Validate functionality of generate_reports() when a bad data is loaded from file."""
```
Comment (dav3r) [inline tests/test_tpt_reports.py]: Slight edit:
```suggestion
    """Validate functionality of generate_reports() when bad data is loaded from a file."""
```

File: tests/test_tpt_reports.py [hunk 4]
```diff
@@ -143,6 +144,40 @@ def test_domain_validation(mock_generate_reports):
...
+    mock_report_gen.return_value = False
+    return_val = tpt_reports.tpt_reports.generate_reports(
+        "test", "test", "cisa.gov", "./test_output", TEST_BAD_JSON_FILE
+    )
+    assert (
+        return_val is False
+    ), "generate_reports() returned a boolean True when a bad data was provided."
```
Comment (dav3r) [inline tests/test_tpt_reports.py]: Similar to above:
```suggestion
    ), "generate_reports() failed to return a boolean False when malformed JSON data was provided."
```

**Review by dav3r — CHANGES_REQUESTED** [review]
I suggested a few small wording changes here.

File: tests/test_tpt_reports.py (hunk 1 above)
Comment (JCantu248) [reply tests/test_tpt_reports.py]: Suggestion applied! Thanks!

File: tests/test_tpt_reports.py (hunk 2 above)
Comment (JCantu248) [reply tests/test_tpt_reports.py]: I committed the suggestion. Thank you.

File: tests/test_tpt_reports.py (hunk 3 above)
Comment (JCantu248) [reply tests/test_tpt_reports.py]: Committed suggestions. Thanks.

Comment (dav3r) [reply tests/test_tpt_reports.py]: For future reference, if you accept a code suggestion, there is typically no need to write anything here in the conversation since GitHub automatically resolves the conversation when you accept the suggestion.  The only exception would be if there is additional info that you want to put out there.

**Review by dav3r — APPROVED** [review]
Looks good now! 👍 

Note that if your pre-merge and post-merge checklists are empty, then those sections can be deleted.  However, now that @nickviola is starting to version this repo, you should ensure that this PR updates the version appropriately.
